instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/properties/SpringConfigProperties.java

```diff
@@ -29,37 +30,86 @@ public class SpringConfigProperties implements ConfigProperties {
   private final ExpressionParser parser;
   private final OtlpExporterProperties otlpExporterProperties;
   private final OtelResourceProperties resourceProperties;
-  private final PropagationProperties propagationProperties;
   private final ConfigProperties otelSdkProperties;
+  private final ConfigProperties customizedListProperties;
+  private final Map<String, List<String>> listPropertyValues = new HashMap<>();
+
+  static final String DISABLED_KEY = "otel.java.disabled.resource.providers";
+  static final String ENABLED_KEY = "otel.java.enabled.resource.providers";
 
   public SpringConfigProperties(
       Environment environment,
       ExpressionParser parser,
       OtlpExporterProperties otlpExporterProperties,
       OtelResourceProperties resourceProperties,
-      PropagationProperties propagationProperties,
+      OtelSpringProperties otelSpringProperties,
       ConfigProperties otelSdkProperties) {
     this.environment = environment;
     this.parser = parser;
     this.otlpExporterProperties = otlpExporterProperties;
     this.resourceProperties = resourceProperties;
-    this.propagationProperties = propagationProperties;
     this.otelSdkProperties = otelSdkProperties;
+    this.customizedListProperties =
+        createCustomizedListProperties(otelSdkProperties, otelSpringProperties);
+
+    listPropertyValues.put(ENABLED_KEY, otelSpringProperties.getJavaEnabledResourceProviders());
+    listPropertyValues.put(DISABLED_KEY, otelSpringProperties.getJavaDisabledResourceProviders());
+    listPropertyValues.put(
+        "otel.experimental.metrics.view.config",
+        otelSpringProperties.getExperimentalMetricsViewConfig());
+    listPropertyValues.put(
+        "otel.experimental.resource.disabled.keys",
+        otelSpringProperties.getExperimentalResourceDisabledKeys());
+    listPropertyValues.put("otel.propagators", otelSpringProperties.getPropagators());
+  }
+
+  private static Map<String, String> addList(
+      String key, List<String> springList, ConfigProperties configProperties) {
+    if (!springList.isEmpty()) {
+      return Collections.singletonMap(key, String.join(",", springList));
+    } else {
+      String otelList = configProperties.getString(key);
+      if (otelList != null) {
+        return Collections.singletonMap(key, otelList);
+      }
+    }
+    return Collections.emptyMap();
+  }
+
+  private static ConfigProperties createCustomizedListProperties(
+      ConfigProperties configProperties, OtelSpringProperties otelSpringProperties) {
+    // io.opentelemetry.instrumentation.resources.ResourceProviderPropertiesCustomizer
```

Comment:
This comment is not clear to me. Could you please expand more?

Reply:
relevant part is here: 

https://github.com/open-telemetry/opentelemetry-java/blob/325822ce8527b83a09274c86a5123a214db80c1d/sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java#L634-L641

spring properties get applied with `configPropertiesCustomizer.apply(properties)` - after `propertiesCustomizers` 3 lines above